fdmt/run.py

```diff
@@ -0,0 +1,193 @@
+'''Runs FDMT on filterbank data, yielding output in the format of the user's
+choice.'''
+
+
+def write_inffile(basename, dm, **infdict):
+    '''Writes a presto .inf file with prefix basename.'''
+    inf_template = (
+""" Data file name without suffix          =  {}
+ Telescope used                         =  {}
+ Instrument used                        =  {}
+ Object being observed                  =  {}
+ J2000 Right Ascension (hh:mm:ss.ssss)  =  {}
+ J2000 Declination     (dd:mm:ss.ssss)  =  {}
+ Data observed by                       =  UNKNOWN
+ Epoch of observation (MJD)             =  {:5.15f}
+ Barycentered?           (1=yes, 0=no)  =  {}
+ Number of bins in the time series      =  {}
+ Width of each time series bin (sec)    =  {:.15g}
+ Any breaks in the data? (1 yes, 0 no)  =  0
+ Orbit removed?          (1=yes, 0=no)  =  {}
+ Dispersion measure (cm-3 pc)           =  {:f}
+ Central freq of low channel (Mhz)      =  {:f}
+ Total bandwidth (Mhz)                  =  {:f}
+ Number of channels                     =  {}
+ Channel bandwidth (Mhz)                =  {}
+ Data analyzed by                       =  {}
+ Any additional notes:
+     fdmt
+""")
+
+    filename = f"{basename}_DM{dm:.2f}.inf"
+    with open(filename, 'w') as ff:
+        ff.write(inf_template.format(
+            filename[:-4],  # data filename without suffix
+            infdict.get('telescope', 'Other'),
+            infdict.get('instrument', 'Other'),
+            infdict.get('source_name', 'Object'),
+            infdict.get('src_raj', 0).to_string(sep=':'),
+            infdict.get('src_dej', 0).to_string(sep=':'),
+            infdict.get('tstart', 0.0),
+            1 if infdict.get('barycentric', False) else 0,
+            1,  # num bins in time series
+            infdict['nsamples'] * infdict['tsamp'],  # width of tseries bin
+            1 if infdict.get('orbit_removed', False) else 0,
+            dm,
+            infdict['fch1'],
+            infdict['foff'] * infdict['nchans'],   # total bandwidth
+            infdict['nchans'],
+            infdict['foff'],
+            'user'
+        ))
+
+
+
+def write_dat_inf_file(basename, data, dm, **infdict):
+    '''Writes a presto .dat and a .inf file with prefix basename.'''
+    # .dat file
+    datname = f"{basename}_DM{dm:.2f}.dat"
```

Comment:
(because I recently ran into dm-precision issues elsewhere) would you min making the dm precision a variable? E.g have a `dmprec=2` argument or something similar and I believe the fstring would then be `f"{basename}_DM{dm:.{dmprec}f}.dat"`

Reply:
yay! thanks :)